src/figuresClasses.ts

```diff
@@ -1,11 +1,102 @@
-export interface Figure {}
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
+
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  shape: Shape;
+
+  color: Color;
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(color: Color, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error("Sides can't form a triangle");
+    }
+    this.shape = Shape.Triangle;
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
[CODE KNOWLEDGE] - When declaring constructor parameters with an access modifier, TypeScript implicitly creates and assigns properties with the same names. You can simplify the constructor by using the parameter properties feature.